File: trains-gdx-core/src/main/java/com/github/caay2000/trains/SimpleTest.java
package com.github.caay2000.trains;

import com.badlogic.gdx.ApplicationListener;
import com.badlogic.gdx.Gdx;
import com.github.caay2000.trains.render.WorldRender;
import com.github.caay2000.trains.world.World;

import static com.github.caay2000.trains.world.WorldConfiguration.worldConfiguration;

public class SimpleTest implements ApplicationListener {

    private World world;
    private WorldRender worldRender;

    @Override
    public void create() {

        world = new World(worldConfiguration()
                .withNumberOfCities(30)
                .withMaxDistanceBetweenCities(150)
                .withMinDistanceBetweenCities(100)
                .withMaxRouteDistance(200)
        );
        worldRender = new WorldRender();
    }

    @Override
    public void resize(int i, int i1) {

    }

    @Override
    public void render() {

        float delta = Gdx.graphics.getDeltaTime();
        world.update(delta);
        worldRender.render(world);
    }


    @Override
    public void pause() {

    }

    @Override
    public void resume() {

    }

    @Override
    public void dispose() {

    }
}

File: trains-gdx-core/src/main/kotlin/com/github/caay2000/trains/SimpleTest.kt
package com.github.caay2000.trains

// import com.github.caay2000.trains.world.object.entity.TrainEventHandler
// import com.github.caay2000.trains.world.object.entity.TrainEventProcessor
import com.github.caay2000.trains.render.WorldRender
import com.github.caay2000.trains.world.GlobalData
import com.github.caay2000.trains.world.World
import com.github.caay2000.trains.world.company.Company
import com.github.caay2000.trains.world.generator.WorldGenerator
import ktx.app.KtxApplicationAdapter

class SimpleTest : KtxApplicationAdapter {

    private lateinit var world: World
    private lateinit var worldRender: WorldRender

    override fun create() {

        Configuration.numberOfCities = 30

        // TrainEventHandler(TrainEventProcessor())
        //Gdx.app.logLevel = Application.LOG_DEBUG

        world = WorldGenerator.generate()
        worldRender = WorldRender()
        for (i in 1..Configuration.numberOfCompanies) {
            world.addCompany(Company(CompanyColors.valueOf(i)))
        }

        GlobalData.world = world
    }

    override fun resize(i: Int, i1: Int) {}
    override fun render() {

        GlobalData.update()



        world.update()
        worldRender.render(world)
    }

    override fun pause() {}
    override fun resume() {}
    override fun dispose() {}
}